rclcpp/src/rclcpp/graph_listener.cpp

```diff
@@ -80,7 +79,7 @@ GraphListener::start_if_not_started()
       0,  // number_of_clients
       0,  // number_of_services
       0,  // number_of_events
-      interrupt_guard_condition_context_.get(),
+      parent_context->get_rcl_context().get(),
```

Comment:
Is it possible for `parent_context->get_rcl_context()` to return `nullptr` here? Should we not check that?

Reply:
Good question, I think it isn't possible except if the `parent_context` was never inited.
If it was inited once, it should still not be a `nullptr` after a call to `shutdown` (as shutdown doesn't clear the rcl context).

The rcl functions are already checking if the context is not null and valid, and returning an approppiate error in those cases.
I think that's enough. If you want I can add another check here, maybe just in the constructor, or in all places that `get_rcl_context` is called.

----

That comment makes me realize that I didn't check if the `parent context` wasn't null after locking.
Adding that.

Reply:
> That comment makes me realize that I didn't check if the parent context wasn't null after locking.
Adding that.

Did you add that? I don't see any new commits.

Reply:
Sorry, I forgot to push the changes :man_facepalming:.